src/OpenLoco/Ui/WindowManager.cpp

```diff
@@ -42,9 +42,12 @@ namespace OpenLoco::Ui::WindowManager
     static loco_global<uint16_t, 0x0052338C> _tooltipNotShownTicks;
     static loco_global<uint16_t, 0x00508F10> __508F10;
     static loco_global<Gfx::Context, 0x0050B884> _screenContext;
+
     static loco_global<uint16_t, 0x00523390> _toolWindowNumber;
     static loco_global<Ui::WindowType, 0x00523392> _toolWindowType;
-    static loco_global<uint16_t, 0x00523394> _toolWidgetIdx;
+    static loco_global<Ui::CursorId, 0x00523393> _currentToolCursor;
+    static loco_global<int16_t, 0x00523394> _toolWidgetIndex;
```

Comment:
The four fields are right next to each other in memory so we could create a struct that holds the tool state. Might simplify things a bit although from how it's used perhaps not.

Reply:
I like that idea. FWIW these 4 fields I did here are just a couple from a much larger section of input/tool related variables that are all related and all adjacent in memory, eg some tooltip stuff just before at 0x0052338A, 0x0052338C, etc so they should be included in this struct as well.

The things directly after it at 0x00523396 relate to scrolling (the map?) and I think aren't related enough to include in this struct. I'll look at adding a struct in the next commit.

Reply:
I'd like tools to become one standalone thing. Tooltips although they have tool in the name i don't see as being part of the same thing

Reply:
I mean it could be all in some super class as well

Reply:
I think this idea still has merit. This PR introduces four pairs of getters and setters to `WindowManager.cpp`. I think ideally we'd reduce this to one getter and setter, passing a ref to a struct around instead.

@duncanspumpkin Thoughts on the above? Follow-up PR, or left for this one?

Reply:
do it in a follow on. This pr has been open for too long already.